src/components/shared/WithSize.js

```diff
@@ -14,6 +15,20 @@ type State = {|
 
 export type SizeProps = $ReadOnly<State>;
 
+const debounce = function(func, wait, scope, ...args) {
```

Comment:
Can you use `lodash.debounce` instead ? You can install it using `yarn add lodash.debounce`. Here is the documentation to use it after you import it: https://lodash.com/docs/4.17.10#debounce

Then it would be nice to have a flow type for this function, you can add it to `src/types/libdef/npm-custom/lodash.debounce_vx.x.x.js`:
```js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
// @flow
// This definition was adapted from lodash-es definition in 
// https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/lodash-es_v4.x.x/flow_v0.63.x-/lodash-es_v4.x.x.js
declare type $$lodash$$DebounceOptions = {
  leading?: boolean,
  maxWait?: number,
  trailing?: boolean
};

declare module "lodash.debounce" {
  declare export default function debounce<F: Function>(func: F, wait?: number, options?: $$lodash$$DebounceOptions): F;
}
```
I didn't try it but hopfully this should work.